src/elm/Model.elm

```diff
@@ -39,3 +40,43 @@ initial =
         , Resource.build "Bicycle" 47 1.07 (Currency 12000)
         ]
     }
+
+
+purchaseResource : Int -> Resource.Resource -> Model -> Model
+purchaseResource count resource model =
+    let
+        transaction =
+            Resource.purchase count resource
+
+        newResource =
+            Resource.applyTransaction transaction
+
+        newResources =
+            replaceResource resource newResource model.resources
+
+        finalCost =
+            Resource.transactionCost transaction
+    in
+    if model |> canPayFor finalCost then
+        { model
+            | resources = newResources
+            , availableFunds = Currency.subtract finalCost model.availableFunds
+        }
+    else
+        model
+
+
+replaceResource : Resource.Resource -> Resource.Resource -> List Resource.Resource -> List Resource.Resource
+replaceResource oldResource newResource =
+    List.map
+        (\r ->
+            if r == oldResource then
+                newResource
+            else
+                r
+        )
```

Comment:
Thoughts on making this function just be the conditional and move the `List.map` up into `newResources`?